pkgs/os-specific/darwin/apple-source-releases/default.nix

```diff
@@ -135,125 +138,135 @@ let
     };
   };
 
-  fetchApple = version: sha256: name: let
+  fetchApple = pname: version: sha256: let
     # When cross-compiling, fetchurl depends on libiconv, resulting
     # in an infinite recursion without this. It's not clear why this
     # worked fine when not cross-compiling
-    fetch = if name == "libiconv"
+    fetch = if pname == "libiconv"
       then stdenv.fetchurlBoot
       else fetchurl;
   in fetch {
-    url = "http://www.opensource.apple.com/tarballs/${name}/${name}-${versions.${version}.${name}}.tar.gz";
+    url = "http://www.opensource.apple.com/tarballs/${pname}/${pname}-${version}.tar.gz";
     inherit sha256;
   };
 
-  appleDerivation_ = name: version: sha256: attrs: stdenv.mkDerivation ({
-    inherit version;
-    name = "${name}-${version}";
+  fetchAppleOnSDK = sdkName: sha256: pname: let
+    version = sdkNames.${sdkName}.${pname};
+  in fetchApple pname version sha256;
+
+  appleDerivation_ = pname: version: sdkName: sha256: attrs: stdenv.mkDerivation ({
+    name = "${pname}-${version}-${sdkName}";
```

Comment:
I prefer using the full name `${pname}-${version}-${sdkPname}-${sdkVersion}` like `xnu-7195.50.7.100.1-macos-11.0.1`, because it's clear and some packages may be in multiple SDKs, say `bootstrap_cmds-121-macos-11.0.1` and `bootstrap_cmds-116-developer-tools-11.3.1`, we can distinguish by name easily.

Reply:
It's just that it seems like a "bootstrap_cmds-121-foo-123" by default is not going to differ in contents from, say, "bootstrap_cmds-121-bar-321" except for the name

Reply:
What about now the derivation have all pname, version and name?

Reply:
Well, defining `name` creates an inconsistency: `(builtins.parseDrvName name).version != version`